detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/GradleInspectorExtractor.java

```diff
@@ -57,9 +57,21 @@ public Extraction extract(File directory, ExecutableTarget gradleExe, @Nullable
             gradleRunner.runGradleDependencies(directory, gradleExe, gradleInspector, gradleCommand, proxyInfo, outputDirectory);
 
             File rootProjectMetadataFile = fileFinder.findFile(outputDirectory, "rootProjectMetadata.txt");
-            List<File> reportFiles = fileFinder.findFiles(outputDirectory, "*_dependencyGraph.txt");
             List<CodeLocation> codeLocations = new ArrayList<>();
 
+            List<File> reportFiles = new ArrayList<>();
+
+            int depthCount = 0;
+            while(true) {
+                String pattern = "_depth" + depthCount;
+                List<File> tempList = fileFinder.findFiles(outputDirectory,"*"+pattern+"_dependencyGraph.txt");
+                if(tempList.isEmpty()) {
+                    break;
+                }
+                reportFiles.addAll(tempList);
+                depthCount++;
+            }
```

Comment:
Instead of getting all the files at once, we would group the files together by their depth and go on the hierarchy until the depth where we have no files, this ensures files are parsed in correct order and those with lower are placed ahead in the list.